apps/crn-server/test/data-providers/contentful/analytics.data-provider.test.ts

```diff
@@ -0,0 +1,1365 @@
+
+          expect(result.items[0]!.workingGroupLeadershipRoleCount).toBe(0);
+        });
+
+        test('Should return workingGroupLeadershipRoleCount of 2 when two users are leaders of two different working groups', async () => {
+          const contentfulGraphQLResponse = getAnalyticsTeamLeadershipQuery();
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items[0]!.linkedFrom!.workingGroupLeadersCollection!.items[0]!.linkedFrom!.workingGroupsCollection!.items =
+            [
+              {
+                sys: {
+                  id: 'working-group-id-1',
+                },
+              },
+            ];
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items.push(
+            {
+              alumniSinceDate: null,
+              linkedFrom: {
+                workingGroupLeadersCollection: {
+                  items: [
+                    {
+                      linkedFrom: {
+                        workingGroupsCollection: {
+                          items: [
+                            {
+                              sys: {
+                                id: 'working-group-id-2',
+                              },
+                            },
+                          ],
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+          );
+          contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+            contentfulGraphQLResponse,
+          );
+
+          const result = await analyticsDataProvider.fetchTeamLeaderShip({});
+
+          expect(result.items[0]!.workingGroupLeadershipRoleCount).toBe(2);
+        });
+
+        test('Should return workingGroupLeadershipRoleCount of 2 when two users are leaders of 3 working groups two of which are unique', async () => {
+          const contentfulGraphQLResponse = getAnalyticsTeamLeadershipQuery();
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items[0]!.linkedFrom!.workingGroupLeadersCollection!.items[0]!.linkedFrom!.workingGroupsCollection!.items =
+            [
+              {
+                sys: {
+                  id: 'working-group-id-1',
+                },
+              },
+            ];
+          contentfulGraphQLResponse.teamsCollection!.items[0]!.linkedFrom!.teamMembershipCollection!.items[0]!.linkedFrom!.usersCollection!.items.push(
+            {
+              alumniSinceDate: null,
+              linkedFrom: {
+                workingGroupLeadersCollection: {
+                  items: [
+                    {
+                      linkedFrom: {
+                        workingGroupsCollection: {
```

Comment:
what if the working group is complete 
![Screenshot 2024-03-05 at 08 14 05](https://github.com/yldio/asap-hub/assets/16595804/19f5be9b-3323-4afd-923f-218d5c35ce4e)


or the user is inactive on the working group?
![Screenshot 2024-03-05 at 08 14 21](https://github.com/yldio/asap-hub/assets/16595804/10cfe2f6-095f-4877-93c4-4105fb7445bd)